django/db/models/fields/related_lookups.py

```diff
@@ -89,6 +89,8 @@ def as_sql(self, compiler, connection):
                     # where place is a OneToOneField and the primary key of
                     # Restaurant.
                     target_field = self.lhs.field.name
+                elif getattr(self.lhs.field.target_field, 'primary_key', False):
```

Comment:
Does delegating to `super()` by adding a `not getattr(self.lhs.field.target_field, 'primary_key', False)` to the `if not getattr(self.rhs, 'has_select_fields', True)` instead also works? I think it would make more sense to let `In.as_sql` deal with this case.